sdk-core/src/main/java/care/data4life/sdk/migration/RecordCompatibilityService.kt

```diff
@@ -19,138 +19,80 @@ package care.data4life.sdk.migration
-    ): MutableList<String> {
-        return tagCryptoService.encryptList(
-            annotations,
-            tagEncryptionKey,
-            ANNOTATION_KEY + TaggingContract.DELIMITER
-        )
-    }
-
-    private fun needsDoubleCall(
-        encodedAndEncryptedTags: List<String>,
-        encryptedTags: List<String>
-    ): Boolean = encodedAndEncryptedTags.sorted() != encryptedTags.sorted()
-
-    override fun searchRecords(
-        alias: String,
-        userId: String,
-        startDate: String?,
-        endDate: String?,
-        pageSize: Int,
-        offSet: Int,
-        tags: Tags,
-        annotations: Annotations
-    ): Observable<List<EncryptedRecord>> {
-        val (encodedAndEncryptedTags, encryptedTags) = encrypt(tags, annotations)
-        return if (needsDoubleCall(encodedAndEncryptedTags, encryptedTags)) {
-            Observable.zip(
-                apiService.searchRecords(
-                    alias,
-                    userId,
-                    startDate,
-                    endDate,
-                    pageSize,
-                    offSet,
-                    encodedAndEncryptedTags.joinToString(",")
-                ),
-                apiService.searchRecords(
-                    alias,
-                    userId,
-                    startDate,
-                    endDate,
-                    pageSize,
-                    offSet,
-                    encryptedTags.joinToString(",")
-                ),
-                BiFunction<List<EncryptedRecord>, List<EncryptedRecord>, List<EncryptedRecord>> { records1, records2 ->
-                    mutableListOf<EncryptedRecord>().also {
-                        it.addAll(records1)
-                        it.addAll(records2)
-                    }
-                }
+    ): List<List<String>> {
+        return annotations.map { annotation ->
+            Pair(
+                ANNOTATION_KEY + DELIMITER,
+                compatibilityEncoder.encode(annotation).copy(second = annotation)
+            )
+        }.map { encodedTagGroup ->
+            encryptTags(
+                tagEncryptionKey,
+                encodedTagGroup.first,
+                encodedTagGroup.second
             )
-        } else {
-            apiService.searchRecords(
-                alias,
-                userId,
-                startDate,
-                endDate,
-                pageSize,
-                offSet,
-                encodedAndEncryptedTags.joinToString(",")
-            ) as Observable<List<EncryptedRecord>>
         }
     }
 
-    override fun countRecords(
-        alias: String,
-        userId: String,
+    override fun resolveSearchTags(
         tags: Tags,
-        annotations: Annotations
-    ): Single<Int> {
-        val (encodedAndEncryptedTags, encryptedTags) = encrypt(tags, annotations)
+        annotation: Annotations
+    ): NetworkingContract.SearchTags {
+        val builder = searchTagsBuilderFactory.newBuilder()
```

Comment:
@jose-cuadros-data4life -> this is solved now, right?

Reply:
Yeah, it looks good to me ✅ Thank you!